src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -843,7 +1048,17 @@ public static string ConstructPrimaryKeyRoute(RestRequestContext context, Dictio
                 sqlMetadataProvider.TryGetExposedColumnName(entityName, primaryKey, out string? pkExposedName);
                 newPrimaryKeyRoute.Append(pkExposedName);
                 newPrimaryKeyRoute.Append("/");
-                newPrimaryKeyRoute.Append(entity[pkExposedName!]!.ToString());
+
+                if (!dbOperationResultSet.ContainsKey(pkExposedName!))
```

Comment:
nit: if you're using null forgiving operator, make sure it is consistent in this foreach block. a few lines above this needs it too.

Reply:
Updated to throw an exception when exposed column name turns out to be `null` (The `TryGetExposedColumn` method returns the original column name even no alias is defined.  The method returns `null` when the data structure used underneath does not contain the primary key entry but this is unlikely as the primary key info gets populated during runtime). However adding in the logic to throw exceptions enables avoiding the usage of null forgiving operator.

Reply:
with an exception, we'll fail the request with 5XX even when the db operation completes successfully. I wonder if we should fail gracefully here since this is just a failure to create the location header.

Reply:
Updated to return an empty string instead of throwing an exception